writers/writer.go

```diff
@@ -0,0 +1,50 @@
+package writers
+
+import (
+	"fmt"
+
+	"github.com/go-kit/kit/metrics"
+	"github.com/gogo/protobuf/proto"
+	"github.com/mainflux/mainflux"
+	log "github.com/mainflux/mainflux/logger"
+	nats "github.com/nats-io/go-nats"
+)
+
+const senML = "out.senml"
+
+type consumer struct {
+	nc     *nats.Conn
+	logger log.Logger
+	name   string
+	repo   MessageRepository
+}
+
+func (c *consumer) handleMsg() nats.MsgHandler {
```

Comment:
Please rename to `consume`. Also, no need for returning the function, just use the same approach as in `normalizer`.